### ISSUE TITLE
Embed url input field misses an associated label

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#1409]
All form controls need to be properly labeled.

The current rendered markup for an embed url field is:
```
<form>
  <input type="url" value="" class="components-placeholder__input" placeholder="Enter URL to embed here…">
  <button type="submit" class="components-button button button-large">Embed</button>
</form>
```

Visually:

<img width="407" alt="screen shot 2017-06-23 at 17 39 28" src="https://user-images.githubusercontent.com/1682452/27489479-f9a332c8-583a-11e7-8e9e-b35fd1f547b3.png">

One option could be to simply use the visible text (in the example above: "Cloudup URL") as label for the input field. 
Worth reminding, according to the WordPress accessibility coding standard, labels should be explicitly associated (use for/id attributes and not wrapping the field). See https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/#labeling
